Source/DafnyTestGeneration/DafnyInfo.cs

```diff
@@ -63,37 +377,88 @@ private class DafnyInfoExtractor : BottomUpVisitor {
           Visit(classDecl);
         } else if (d is IndDatatypeDecl datatypeDecl) {
           Visit(datatypeDecl);
+        } else if (d is NewtypeDecl newTypeDecl) {
```

Comment:
I'd love to move `Visit(TopLevelDecl d)` up to the base `BottomUpVisitor` while we're at it. I've been annoyed previously that the code visitor implementation doesn't cover the whole AST. :)

Reply:
This makes sense! Will move this bit and will see if I can draft a separate PR for any other missing branches I might find)